### PR TITLE
Additional Snowflake Parameters

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

Added Role and Warehouse as additional connection parameters. Added a struct iterator

### Select type(s) of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: provider/snowflake.go [hunk 1]
```diff
@@ -27,6 +33,8 @@ type SnowflakeConfig struct {
 	Account        string
 	Database       string
 	Schema         string
+	Warehouse      string
```
Comment (simba-git) [inline provider/snowflake.go:36]: I recommend adding a struct tag:
\`snowflake:warehouse\`

File: provider/snowflake.go [hunk 2]
```diff
@@ -27,6 +33,8 @@ type SnowflakeConfig struct {
...
+	Role           string
```
Comment (simba-git) [inline provider/snowflake.go:37]: struct tag
\`snowflake:role\`

File: provider/snowflake.go [hunk 3]
```diff
@@ -19,6 +20,11 @@ const (
 	sfTimestamp                     = "TIMESTAMP_NTZ"
 )
 
+var snowflakeParameterAlias = map[string]string{
```
Comment (simba-git) [inline provider/snowflake.go:23]: Why are only these two non-capital?

File: helpers/struct_iterator/struct_iterator.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		return si.Next()
+	}
+	return false
+}
+
+func (si *StructIterator) inRange() bool {
+	return si.idx < si.numField
+}
+
+func (si *StructIterator) ItemName() string {
```
Comment (simba-git) [inline helpers/struct_iterator/struct_iterator.go:29]: Remove ItemName, call it Name

File: helpers/struct_iterator/struct_iterator.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	return si.val.Type().Field(si.idx).Name
+}
+
+func (si *StructIterator) ItemValue() interface{} {
```
Comment (simba-git) [inline helpers/struct_iterator/struct_iterator.go:33]: Remove ItemValue call it Value

File: helpers/struct_iterator/struct_iterator.go (hunk 4 above)
Comment (simba-git) [reply helpers/struct_iterator/struct_iterator.go:29]: Call it Field or Key actually

File: helpers/struct_iterator/struct_iterator.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	return si.val.Field(si.idx).Interface()
+}
```
Comment (simba-git) [inline helpers/struct_iterator/struct_iterator.go:35]: Add a new method call is

Tag(string)
https://pkg.go.dev/reflect#StructTag.Get

File: provider/snowflake.go (hunk 3 above)
Comment (sdreyer) [reply provider/snowflake.go]: These two are used in the parameters, created the struct iterator so we can add others if needed but these seem to be the most common. The godoc says they're lowercase.

File: helpers/struct_iterator/struct_iterator.go (hunk 6 above)
Comment (sdreyer) [reply helpers/struct_iterator/struct_iterator.go:35]: Used Lookup instead for more expected results

File: helpers/struct_iterator/struct_iterator.go [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+func (si *StructIterator) Key() string {
+	return si.val.Type().Field(si.idx).Name
+}
+
+func (si *StructIterator) Value() interface{} {
+	return si.val.Field(si.idx).Interface()
+}
+
+// Tag Fetches the struct tag for the current key. If the
+// specified tag does not exist, returns nil
```
Comment (simba-git) [inline helpers/struct_iterator/struct_iterator.go:38]: returns ""

File: helpers/struct_iterator/struct_iterator_test.go [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+func TestNewStructIterator(t *testing.T) {
+	type emptyStruct struct{}
+
+	type args struct {
+		s interface{}
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *StructIterator
```
Comment (simba-git) [inline helpers/struct_iterator/struct_iterator_test.go:17]: Could we add one tag test

**Review by simba-git — APPROVED** [review]
Two quick notes, then merge

**Pull request merged**